### DISCUSSION TITLE
Is it possible to find keywords positions?

### DISCUSSION THREAD

**Discussion by Artawower** [rasendubi/uniorg#49] [Q&A]
May I ask why some nodes do not have `contentsBegin` and `contentsEnd`? Is this part of the specification? And is there any way to enable this feature?

I'm trying to embellish keywords by range, but now I have to calculate their position by their neighbors. And I have no idea how to do it better.

**Comment by rasendubi** [comment]
`contentsBegin`/`contentsEnd` are only present in greater elements (elements that contain other elements) and they actually define the range where sub-elements are parsed. Keywords are simple elements (not greater ones), so they should not have `contentsBegin`/`contentsEnd`. (This behavior is consistent with org-element.el parsing.)

Though, they should have simple `begin`/`end` fields. These are currently missing because they weren't a priority (I focused on getting org->html conversion working and positions aren't useful for that).

If you want to work on that, I am willing to accept this support. It should be relatively straightforward to add: most of `parseXxx` functions already have `begin`/`end` calculated, we just have to add these when calling to `u` (see `packages/uniorg-parser/src/parser.ts`); and add appropriate typings in `packages/uniorg/src/index.ts`.
